lib/matplotlib/projections/polar.py

```diff
@@ -1447,12 +1447,23 @@ def format_sig(value, delta, opt, fmt):
                     cbook._g_sig_digits(value, delta))
             return f"{value:-{opt}.{prec}{fmt}}"
 
-        return ('\N{GREEK SMALL LETTER THETA}={}\N{GREEK SMALL LETTER PI} '
-                '({}\N{DEGREE SIGN}), r={}').format(
+        # In case fmt_xdata was not specified, resort to default
+        if self.fmt_xdata is None:
+            return ('\N{GREEK SMALL LETTER THETA}={}\N{GREEK SMALL LETTER PI} '
+                    '({}\N{DEGREE SIGN}), r={}').format(
                     format_sig(theta_halfturns, delta_t_halfturns, "", "f"),
                     format_sig(theta_degrees, delta_t_degrees, "", "f"),
-                    format_sig(r, delta_r, "#", "g"),
+                    format_sig(r, delta_r, "#", "g")
+                    if self.fmt_ydata is None
+                    else self.format_ydata(r)
                 )
+        else:
+            return '\N{GREEK SMALL LETTER THETA}={}, r={}'.format(
+                        self.format_xdata(theta),
+                        format_sig(r, delta_r, "#", "g")
+                        if self.fmt_ydata is None
```

Comment:
This is always false and can be removed, because you checked in l. 1451.

Reply:
Why is it always false? Like above, we can do set `ax.fmt_ydata = function` and `self.fmt_xdata` will be None while `self.fmt_ydata` won't. (In the if-statement we only check `fmt_xdata`).